src/pybind/mgr/cephadm/module.py

```diff
@@ -2402,7 +2405,10 @@ def _create_daemons(self, daemon_type, spec, daemons,
             if daemon_type == 'mon':
                 args.append((daemon_id, host, network))  # type: ignore
             else:
-                args.append((daemon_id, host))  # type: ignore
+                if extra_config:
+                    args.append((daemon_id, host, extra_config))  # type: ignore
+                else:
+                    args.append((daemon_id, host))  # type: ignore
```

Comment:
```suggestion
            elif daemon_type = 'iscsi'
                 args.append((daemon_id, host, spec))  # type: ignore
            else:
                args.append((daemon_id, host))  # type: ignore
```

You could pass the spec through and use it during _config_iscsi